test/Microsoft.ML.TestFramework/TestSparseDataView.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ML.Runtime.Command;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.Tools;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Runtime.RunTests
+{
+    public sealed partial class TestSparseDataView : TestDataViewBase
+    {
+        const string Cat = "DataView";
+
+        public TestSparseDataView(ITestOutputHelper obj) : base(obj)
+        {
+        }
+
+        class DenseExample<T>
+        {
+            [VectorType(2)]
+            public T[] X;
+        }
+
+        class SparseExample<T>
+        {
+            [VectorType(5)]
+            public VBuffer<T> X;
+        }
+
+        [Fact]
+        [TestCategory(Cat)]
+        public void SparseDataView()
+        {
+            GenericSparseDataView(new[] { 1f, 2f, 3f }, new[] { 1f, 10f, 100f });
+            GenericSparseDataView(new DvInt4[] { 1, 2, 3 }, new DvInt4[] { 1, 10, 100 });
+            GenericSparseDataView(new DvBool[] { true, true, true }, new DvBool[] { false, false, false });
+            GenericSparseDataView(new double[] { 1, 2, 3 }, new double[] { 1, 10, 100 });
+            GenericSparseDataView(new DvText[] { new DvText("a"), new DvText("b"), new DvText("c") },
+                                  new DvText[] { new DvText("aa"), new DvText("bb"), new DvText("cc") });
+            Done();
+        }
+
+        void GenericSparseDataView<T>(T[] v1, T[] v2)
+        {
+            var inputs = new[] {
+                new SparseExample<T>() { X = new VBuffer<T> (5, 3, v1, new int[] { 0, 2, 4 }) },
+                new SparseExample<T>() { X = new VBuffer<T> (5, 3, v2, new int[] { 0, 1, 3 }) }
+            };
+            var host = new TlcEnvironment();
+            var data = host.CreateStreamingDataView(inputs);
+            var value = new VBuffer<T>();
+            int n = 0;
+            using (var cur = data.GetRowCursor(i => true))
+            {
+                var getter = cur.GetGetter<VBuffer<T>>(0);
+                while (cur.MoveNext())
+                {
+                    getter(ref value);
+                    Assert.True(value.Count == 3);
+                    ++n;
+                }
+            }
+            Assert.True(n == 2);
+        }
+
+        [Fact]
```

Comment:
>Fact [](start = 9, length = 4)

Thank you for fixing my mess!
Your code definitely do right thing, and it's nice to have tests, but I want to point one thing. Your tests never hit TypedCursor changes.
There is two conversions: one from some collection to IDataview through code in DataViewConstructionUltis, and from IDataVIew to Collection via TypedCursor and your tests cover only first case.

In order to invoke second case you need to do something like:
dataView.AsEnumerable<ClassWithArrays>(env, false).GetEnumerator() and enumerate over it.

Reply:
I'll do it.